server/src/main/java/org/elasticsearch/reservedstate/service/FileSettingsService.java

```diff
@@ -341,6 +333,20 @@ private void watcherThread() {
         }
     }
 
+    // package private for testing
+    void processSettingsAndNotifyListeners() throws InterruptedException {
+        try {
+            processFileSettings(operatorSettingsFile()).get();
+            startupLatch.onResponse(null);
+            for (var listener : eventListeners) {
+                listener.settingsChanged();
+            }
+        } catch (ExecutionException e) {
+            logger.error("Error processing operator settings json file", e.getCause());
+            startupLatch.onFailure((Exception) e.getCause());
```

Comment:
If the cluster ends up running multiple concurrent elections for the first master then it's possible that the first elected master might be usurped shortly after, which could result in an exception here on which I think we should retry rather than just shutting the node down. Use `MasterService#isPublishFailureException` to detect that.